romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -54,66 +54,94 @@ def process(self, input_models):
+        if len(library) < 2:
+            # if input can be parsed into a ModelLibrary
+            # but is not valid then log a warning message and
+            # skip outlier detection step
             self.log.warning(
-                "Skipping outlier_detection - input cannot be parsed into a ModelContainer."
+                "Skipping outlier_detection - at least two imaging observations are needed."
             )
             self.skip = True
-            return input_models
-
-        # validation
-        if len(self.input_models) >= 2 and all(
-            model.meta.exposure.type == "WFI_IMAGE" for model in self.input_models
-        ):
-            # Setup output path naming if associations are involved.
-            asn_id = self.input_models.asn_table.get("asn_id", None)
-            if asn_id is not None:
-                _make_output_path = self.search_attr(
-                    "_make_output_path", parent_first=True
-                )
-                self._make_output_path = partial(_make_output_path, asn_id=asn_id)
-
-            detection_step = outlier_detection.OutlierDetection
-            pars = {
-                "weight_type": self.weight_type,
-                "pixfrac": self.pixfrac,
-                "kernel": self.kernel,
-                "fillval": self.fillval,
-                "nlow": self.nlow,
-                "nhigh": self.nhigh,
-                "maskpt": self.maskpt,
-                "grow": self.grow,
-                "snr": self.snr,
-                "scale": self.scale,
-                "backg": self.backg,
-                "kernel_size": self.kernel_size,
-                "save_intermediate_results": self.save_intermediate_results,
-                "resample_data": self.resample_data,
-                "good_bits": self.good_bits,
-                "allowed_memory": self.allowed_memory,
-                "in_memory": self.in_memory,
-                "make_output_path": self.make_output_path,
-                "resample_suffix": "i2d",
-            }
 
+        # check that all inputs are WFI_IMAGE
+        if not self.skip:
+            with library:
+                # TODO: a more efficient way to check this without opening all models
+                for i, model in enumerate(library):
+                    if model.meta.exposure.type != "WFI_IMAGE":
+                        self.skip = True
+                    library.discard(i, model)
+                if self.skip:
+                    self.log.warning(
+                        "Skipping outlier_detection - all WFI_IMAGE exposures are required."
+                    )
+
+        # if skipping for any reason above...
+        if self.skip:
+            # set meta.cal_step.outlier_detection to SKIPPED
+            with library:
+                for i, model in enumerate(library):
+                    model.meta.cal_step["outlier_detection"] = "SKIPPED"
+                    library[i] = model
+            return library
```

Comment:
Good to duplicate the existing behavior, but FWIW, I think we'd be better off crashing or throwing an exception or ... in the case that someone sends something that's not an image.  Skipping feels more likely to cause problems than crashing does.

Reply:
I opened an issue to track this change: https://github.com/spacetelescope/romancal/issues/1302 If you're rather I make the change in this PR I'm happy to do so.